src/nvim/eval/funcs.c

```diff
@@ -8115,16 +8115,16 @@ static void f_rpcrequest(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     // If this is called from a provider function, restore the scope
     // information of the caller.
     save_current_sctx = current_sctx;
-    save_sourcing_name = sourcing_name;
-    save_sourcing_lnum = sourcing_lnum;
+    save_sourcing_name = SOURCING_NAME;
+    save_sourcing_lnum = SOURCING_LNUM;
```

Comment:
Shouldn't we be pushing to the execution stack in places like this now?

Reply:
Yep, good call - this function is Neovim-specific but we should adapt it as well.

Reply:
@seandewar does that (f1dd6a9) look good?